From c0404f122d5a5c9f0a5ebb39beab596a1ebe1ac3 Mon Sep 17 00:00:00 2001
From: Bruce Dawson <brucedawson@chromium.org>
Date: Fri, 28 May 2021 05:35:20 +0000
Subject: [PATCH] Run generate_bindings.py under Python 3

generate_bindings.py runs under Python 2 due to the "run_under_python2
 = true" statement in its BUILD.gn file. This slows down binding
generation slightly, which is why it was noticed, but it also needs
fixing as part of the Python 3 migration project.

The only incompatibility was adding the result of map() to a list - in
Python 3 that needs an explicit conversion.

Switching to Python 3 drops the run time of this script by about 20%,
which is a nice bonus considering how much CPU time it uses.

Bug: 1194277, 1214033
Change-Id: I2d9cb836d153c775dc0ec912237bb9e1b9cf4cd0
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2922869
Reviewed-by: Kentaro Hara <haraken@chromium.org>
Commit-Queue: Bruce Dawson <brucedawson@chromium.org>
Cr-Commit-Position: refs/heads/master@{#887468}
---
 src/third_party/blink/renderer/bindings/BUILD.gn | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/third_party/blink/renderer/bindings/BUILD.gn b/src/third_party/blink/renderer/bindings/BUILD.gn
index 385c9d0c45..b9f1c91051 100644
--- a/src/third_party/blink/renderer/bindings/BUILD.gn
+++ b/src/third_party/blink/renderer/bindings/BUILD.gn
@@ -171,9 +171,6 @@ action_with_pydeps("web_idl_database") {
 
 template("generate_bindings") {
   action_with_pydeps(target_name) {
-    # TODO(crbug.com/1194277): Investigate non-determinism in Py3 builds.
-    run_under_python2 = true
-
     script = "${bindings_scripts_dir}/generate_bindings.py"
 
     if (defined(invoker.pool)) {
